application/akka/java/example/src/main/java/com/example/Example.java

```diff
@@ -0,0 +1,50 @@
+package com.example;
+
+import akka.NotUsed;
+import akka.actor.typed.Behavior;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.ServerBinding;
+import akka.http.javadsl.server.Route;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.ActorSystem;
+import akka.http.javadsl.server.AllDirectives;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.CompletionStage;
+
+public class Example extends AllDirectives {
+    final Route root = path("", () -> get(() -> {
+            if (System.getenv("DEMO_USERNAME") == null || System.getenv("DEMO_PASSWORD") == null) {
+                return complete("not all variables are set\n");
+            } else {
+                return complete("Welcome " + System.getenv("DEMO_USERNAME") + "\n");
+            }
+        }));
+
+    static void startHttpServer(Route route, ActorSystem<?> system) {
+        CompletionStage<ServerBinding> futureBinding =
+            Http.get(system).newServerAt("localhost", 8080).bind(route);
```

Comment:
Since we're running these apps in docker containers, I think we should use `"0.0.0.0"` instead of `"localhost"`.

The current code results in `ERR_SOCKET_NOT_CONNECTED` on my laptop.

Reply:
Indeed. it does fail.
It is fixed now. The user can access from the browser.